setup-database.sh

```diff
@@ -2,43 +2,20 @@
 
 source /env-data.sh
 
-
-SETUP_LOCKFILE="${DATADIR}/.postgresql.init.lock"
-
-# This script will setup the necessary folder for database
-chown -R postgres /var/lib/postgresql
 # test if DATADIR has content
-if [[ "${EXISTING_DATA_DIR}" == 'FALSE' ]]; then
-    # Only attempt reinitializations if ${DATADIR} doesn't exists
-    if [[ ! -d "${DATADIR}" && ! -f "${SETUP_LOCKFILE}" ]]; then
-        # No content yet - first time pg is being run!
-        # No Replicate From settings. Assume that this is a master database.
-        # Initialise db
-        echo "Initializing Postgres Database at ${DATADIR}"
-        mkdir -p ${DATADIR}
-        chown -R postgres /var/lib/postgresql
-        su - postgres -c "$INITDB -U postgres -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} -D ${DATADIR}"
-        touch ${SETUP_LOCKFILE}
-    elif [[ ! -f "${SETUP_LOCKFILE}" ]]; then
-        # DATADIR is not empty but no lockfile
-        # DBAdmin need to resolve it manually
-        echo "${DATADIR} exists, but no lockfile."
-        echo "Please resolve by one of this options:"
-        echo "1. Adding a lockfile ${SETUP_LOCKFILE} to indicate that you want to use this existing data directory. Then set EXISTING_DATA_DIR to true."
-        echo "2. Delete ${DATADIR} and set EXISTING_DATA_DIR variable to false"
-        # Hang the process
-        tail -f /dev/null
-    fi
-elif [[ ! -f "${SETUP_LOCKFILE}" ]]; then
-    # DATADIR is not empty but no lockfile
-    # DBAdmin need to resolve it manually
-    echo "${DATADIR} exists, but no lockfile."
-    echo "Please resolve by one of this options:"
-    echo "1. Adding a lockfile ${SETUP_LOCKFILE} to indicate that you want to use this existing data directory. Then set EXISTING_DATA_DIR to true."
-    echo "2. Delete ${DATADIR} and set EXISTING_DATA_DIR variable to false"
-    # Hang the process
-    tail -f /dev/null
+if [[ "${RECREATE_DATADIR}" == 'TRUE' ]]; then
+    # Only attempt reinitializations if ${RECREATE_DATADIR} is true
+    # No Replicate From settings. Assume that this is a master database.
+    # Initialise db
+    echo "Initializing Postgres Database at ${DATADIR}"
+    mkdir -p ${DATADIR}
+    rm -rf ${DATADIR}/*
+    chown -R postgres:postgres ${DATADIR}
+    echo "Initializing with command:"
+    command="$INITDB -U postgres -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} -D ${DATADIR}"
```

Comment:
I thought we could also add an extra env variable for users to add to initialize the data directory ie POSTGRES_INITDB_EXTRA
the default could be empty but users could expand it example https://www.postgresql.org/docs/12/app-initdb.html
ie POSTGRES_INITDB_EXTRA=--waldir=/data
```
command="$INITDB -U postgres -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} -D ${DATADIR} ${POSTGRES_INITDB_EXTRA}"
```
where POSTGRES_INITDB_EXTRA=--waldir=/data

Reply:
Good idea! Might as well do that too.

Reply:
Did you add this option. I think it would also help with the wal being full because you could set it to a different folder